File: frontend/src/voiceCorrections.js
// © 2026 Daniel Partel – ΣIRIUS Assistant. Tous droits réservés. Toute reproduction, modification, distribution ou utilisation non autorisée est strictement interdite. Logiciel protégé par le droit d'auteur (Code de la propriété intellectuelle – France).

export function normalizeVoiceTranscript(text) {
  if (!text) return "";
  return [
    [/\bdaniel\s*[,;:]?\s*pontel\b/gi, "Daniel Partel"],
    [/\bdaniel\s*[,;:]?\s*parti\b/gi, "Daniel Partel"],
    [/\bdaniel\s*[,;:]?\s*part\s+elle\b/gi, "Daniel Partel"],
    [/\bpontel\b/gi, "Partel"],
    [/\bpart\s+elle\b/gi, "Partel"],
    [/\bserious\b/gi, "SIRIUS"],
    [/\bsyrius\b/gi, "SIRIUS"],
    [/\bcirius\b/gi, "SIRIUS"],
    [/\bsirus\b/gi, "SIRIUS"],
    [/\bcyrus\b/gi, "SIRIUS"],
    [/\bs[ée]rieux\b/gi, "SIRIUS"],
    [/\bargousse\b/gi, "ARGUS"],
    [/\bargus\b/gi, "ARGUS"],
    [/\bargue us\b/gi, "ARGUS"],
    [/\batlace\b/gi, "ATLAS"],
    [/\bathlas\b/gi, "ATLAS"],
    [/\boracle divin\b/gi, "Oracle Divin"],
    [/\bnummarius\b/gi, "Nummarius"],
    [/\bnumarius\b/gi, "Nummarius"],
    [/\bth[ée]mis\b/gi, "Thémis"],
    [/\bpant[ée]on\b/gi, "Panthéon"],
    [/\bh[ée]phaistos\b/gi, "Héphaïstos"],
    [/\bh[ée]racl[eè]s\b/gi, "Héraclès"],
    [/\bpythagor[eé]\b/gi, "Pythagore"],
    [/\bcaliop[eé]\b/gi, "Calliope"],
    [/\bprom[ée]th[ée]\b/gi, "Prométhée"],
    [/\bherm[èe]s\b/gi, "Hermès"],
    [/\bagora\b/gi, "Agora"],
    [/\bout look\b/gi, "Outlook"],
    [/\bhot mail\b/gi, "Hotmail"],
  ].reduce((value, [pattern, replacement]) => value.replace(pattern, replacement), String(text));
}

export function chooseBestVoiceTranscript(alternatives) {
  const list = (alternatives || [])
    .map((item) => ({
      text: normalizeVoiceTranscript(item?.text || item?.transcript || ""),
      confidence: Number.isFinite(item?.confidence) ? item.confidence : 0,
    }))
    .filter((item) => item.text.trim());
  if (!list.length) return "";

  const score = (item) => {
    let value = item.confidence;
    if (/\b(SIRIUS|ARGUS|ATLAS|Outlook|Partel)\b/i.test(item.text)) value += 0.35;
    if (/\bdaniel\s+partel\b/i.test(item.text)) value += 0.5;
    return value;
  };

  return list.sort((a, b) => score(b) - score(a))[0].text.trim();
}

File: backend/voice_corrections.py
# © 2026 Daniel Partel – ΣIRIUS Assistant. Tous droits réservés.
"""Corrections ciblées et extensibles des transcriptions vocales fréquentes."""

from __future__ import annotations

import re


def normalize_voice_transcript(text: str) -> str:
    if not text:
        return ""
    corrected = str(text)
    replacements = (
        (r"\bdaniel\s*[,;:]?\s*pontel\b", "Daniel Partel"),
        (r"\bdaniel\s*[,;:]?\s*parti\b", "Daniel Partel"),
        (r"\bdaniel\s*[,;:]?\s*part\s+elle\b", "Daniel Partel"),
        (r"\bpontel\b", "Partel"),
        (r"\bpart\s+elle\b", "Partel"),
        (r"\bserious\b", "SIRIUS"),
        (r"\bsyrius\b", "SIRIUS"),
        (r"\bcirius\b", "SIRIUS"),
        (r"\bsirus\b", "SIRIUS"),
        (r"\bcyrus\b", "SIRIUS"),
        (r"\bs[ée]rieux\b", "SIRIUS"),
        (r"\bargousse\b", "ARGUS"),
        (r"\bargus\b", "ARGUS"),
        (r"\bargue us\b", "ARGUS"),
        (r"\batlace\b", "ATLAS"),
        (r"\bathlas\b", "ATLAS"),
        (r"\boracle divin\b", "Oracle Divin"),
        (r"\bnummarius\b", "Nummarius"),
        (r"\bnumarius\b", "Nummarius"),
        (r"\bth[ée]mis\b", "Thémis"),
        (r"\bpant[ée]on\b", "Panthéon"),
        (r"\bh[ée]phaistos\b", "Héphaïstos"),
        (r"\bh[ée]racl[eè]s\b", "Héraclès"),
        (r"\bpythagor[eé]\b", "Pythagore"),
        (r"\bcaliop[eé]\b", "Calliope"),
        (r"\bprom[ée]th[ée]\b", "Prométhée"),
        (r"\bherm[èe]s\b", "Hermès"),
        (r"\bagora\b", "Agora"),
        (r"\bout look\b", "Outlook"),
        (r"\bhot mail\b", "Hotmail"),
    )
    for pattern, replacement in replacements:
        corrected = re.sub(pattern, replacement, corrected, flags=re.IGNORECASE)
    return corrected
